qiskit_experiments/database_service/db_analysis_result.py

```diff
@@ -143,7 +146,9 @@ def load(cls, result_id: str, service: DatabaseServiceV1) -> "DbAnalysisResultV1
             The loaded analysis result.
         """
         # Load data from the service
-        return cls._from_service_data(service.analysis_result(result_id))
+        instance = cls._from_service_data(service.analysis_result(result_id))
+        instance._created_in_db = True
```

Comment:
The `DbExperimentDataV1.load` method probably also needs to have this attribute set on the object it returns.

Reply:
Good point.

Reply:
Fixed in c6d0a9f0a.